src/Microsoft.Azure.WebJobs/FunctionNameAttribute.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// This is for providing a custom name to for an SDK function.
```

Comment:
That's a good question. I think the limitations we need for that name is that it has to be a valid fileName since, at least short term, that'll get translated into a folder with a function.json inside of it. I don't know how much sense DisplayName would make for that since it'll be in the folderName, http route, etc. also it's going to be a required attribute for an SDK function to be transformed into a function.json.